### ISSUE TITLE
Update position device traker

### ISSUE THREAD

**Issue by manustars** [home-assistant/android#1982]
**Home Assistant Android version:**
2021.10.0-full
**Android version:**
11

**Phone model:**
Samsung s21

**Home Assistant version:**
core-2021.11.5
supervisor-2021.10.8

**Last working Home Assistant release (if known):**

**Description of problem:**
the update position is slow and update only i open the app (background acces is enable and position is enable)

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```

```

**Screenshot of problem:**
![image](https://user-images.githubusercontent.com/60410607/144268742-2e38e570-ab2c-4b9d-904b-4c32d57bd54f.png)

**Additional information:**
on the screenshoot the green line is false becauso im not in home, when i open the app the ha update to "not home"

**Comment by dshokouhi** [comment]
Check all these steps: https://companion.home-assistant.io/docs/troubleshooting/faqs/#location-is-not-updating-in-android-app

If this doesn't work then provide the logs from the device

**Comment by manustars** [comment]
The config is correct, now i have a question:

when i turnoff the smartphone, the app HA not have automatic startup?

**Comment by dshokouhi** [comment]
> when i turnoff the smartphone, the app HA not have automatic startup?

automatic start up yes but nothing we can do if your device decides to still kill the app,  You also have a samsung device so check power settings too, there is more to adjust than just battery optimization on that device. Unfortuantely sasmsung likes to hide a lot of settings.

https://dontkillmyapp.com/samsung

**Comment by cpainchaud** [comment]
You may want to check that you are not using/playing Niantic's Pikmin Bloom game
(https://github.com/home-assistant/android/issues/1925 https://www.reddit.com/r/PikminBloomApp/comments/qqyf0s/warning_the_pikmin_bloom_app_will_interfere_with/)

**Comment by manustars** [comment]
i set the power setting but the same problem, and now recived android 12

**Comment by manustars** [comment]
now i see the sensor update last 10minut, but the positio its wrong

**Comment by dshokouhi** [comment]
@manustars if you want us to continue to troubleshoot please provide the logs as I had asked for earlier.

**Comment by dshokouhi** [comment]
> how can i send logs privately and not publicly?

just remove personally identifiable information, in this case you can mask the GPS coordinates and the URL of your instance.

**Comment by manustars** [comment]
[homeassistant_companion_log_11-16-2021_11-24-36.txt](https://github.com/home-assistant/android/files/7728291/homeassistant_companion_log_11-16-2021_11-24-36.txt)
i dont know delete all

**Comment by dshokouhi** [comment]
```
12-16 11:07:32.140  9791  9826 E LocBroadcastReceiver: Could not update location.
12-16 11:07:32.140  9791  9826 E LocBroadcastReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: javax.net.ssl.SSLHandshakeException: Unacceptable certificate: CN=R3, O=Let's Encrypt, C=US
```

You have some SSL errors to resolve on your end, the application is unable to send data back home.

```
12-16 11:07:32.153  9791  9827 E LocBroadcastReceiver: Caused by: java.security.cert.CertificateExpiredException: Certificate expired at Wed Sep 29 21:21:40 GMT+02:00 2021 (compared to Thu Dec 16 11:07:32 GMT+01:00 2021)
```

Looks like your certificate may have expired???

**Comment by dshokouhi** [comment]
> i use let's encrypct certificate, on my pfsense firewal the certificate is automatic renewal every 3 month, now the new expiration is 06/03/2022

do you still see the same errors?

**Comment by manustars** [comment]
the uploaded log is today

**Comment by dshokouhi** [comment]
@manustars if you did not change anything then maybe you need to restart your host machine or HA service because the error indicates the old expiration date. If you did change something then you need to look at the new logs to see if the SSL error changed.  SSL errors are not ones the app will solve so you will need to do some troubleshooting on your end.

**Comment by dshokouhi** [comment]
@manustars if you made changes to HA then you should retest from the app and check the new logs.  If SSL is not being used then you won't see the SSL error. The logs will indicate which URL is being used be it external or internal.

**Comment by manustars** [comment]
the problem I see it when I open the app and I see that the last position update is 10 hours before, then after a few seconds that I have the app open the position on ha is updated. what do you mean by app test?

**Comment by dshokouhi** [comment]
> the problem I see it when I open the app and I see that the last position update is 10 hours before, then after a few seconds that I have the app open the position on ha is updated.

This really suggests you have more settings to adjust to ensure the app is not being killed by the system. There is nothing we can do if this keeps happening other than to ask you to ensure the app is not being killed.

> what do you mean by app test?

I meant to perform another test once you verified the SSL errors are gone, then if it still doesn't work we need to check the logs again and see what the error is.

**Comment by manustars** [comment]
all setting on https://dontkillmyapp.com/samsung it's applied.

i see on the log this "javax.net.ssl.SSLHandshakeException: Unacceptable certificate: CN=R3, O=Let's Encrypt, C=US"

**Comment by manustars** [comment]
sorry but i don't know other provider generate free certificate. can you recommend one to me?

**Comment by dshokouhi** [comment]
Personally I am using certbot https://certbot.eff.org/

**Comment by manustars** [comment]
I have a doubt, I have not uploaded the certificate to HA but I connect via revers proxy (HAproxy) which forwards https calls to http of ha

**Comment by manustars** [comment]
i resolve issue for ssl certificate, but the problem is same.
[homeassistant_companion_log_11-28-2021_10-55-7.txt](https://github.com/home-assistant/android/files/7784081/homeassistant_companion_log_11-28-2021_10-55-7.txt)

**Comment by dshokouhi** [comment]
Now you have a new SSL error, it seems whoever you selected for the SSL cert android does not like:

```
12-28 10:15:37.951 23158 11420 E WebSocketRepository: javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
```

```
12-28 10:23:44.031 23158 23595 E SensorReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
```

This page should help you understand the issue and what android expects in terms of a SSL certificate.

https://developer.android.com/training/articles/security-ssl#CommonProblems

**Comment by steven52880** [comment]
I have a similar [problem](https://github.com/home-assistant/android/issues/1879)

The web page can be displayed in chrome and the app. The app can update the sensor when it is running in the foreground. But it stop working when I switched it to background. I am sure the app is running in background, because I recieved updates as soon as I change back the certificate in my server.

Now I am using Let's Encrypt's 3 months certificate, but I have to renew it frequently.
I have tried free 1 year ssl certificate provided by Alibabacloud and Tencentcloud, but neither of them works.

**Comment by dshokouhi** [comment]
> Now I am using Let's Encrypt's 3 months certificate, but I have to renew it frequently.
> I have tried free 1 year ssl certificate provided by Alibabacloud and Tencentcloud, but neither of them works.

Android can be picky about the SSL certificate used nothing the app can do to help that as we do not ignore SSL errors, personally I am using certbot to have the cert renew automatically every 60 days and have not had an issue with it.

**Comment by manustars** [comment]
if it can be useful I explain what I did.
I have certified let's encrypt managed by pfsense with automatic renewal, from pfsense I have deleted all the ca let's and the created certificate, I performed the procedure to create a new certificate and automatically entered the 2 ca of let's encrypt, today I went to check via web browser if it had updated the positions with the phone just turned on and without having started the HA app, and it seems that now it updates correctly. I will keep an eye on it to confirm that it works now.
